docs/advanced/experimental/Kubernetes.md

```diff
@@ -0,0 +1,132 @@
+# Cookbook on `Kubernetes` Deployment
+
+Deploy your `Flow` on `Kubernetes`.
+## Requirements
+- please setup a `Kubernetes` cluster and pull the credentials. e.g.:
+  [minikube](https://minikube.sigs.k8s.io/docs/start/) (testing),
+  [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine),
+  [Amazon EKS](https://aws.amazon.com/eks),
+  [Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service),
+  [Digital Ocean](https://www.digitalocean.com/products/kubernetes/)
+  
+## Deploy your `Flow`
+To deploy a `Flow` on `Kubernetes`, you have to set the `infrastructure='K8S'` when creating the `Flow`.
+Call `flow.start()` to trigger the deployment. Make sure all `Executors` are containerized.
+
+## Example
+In this example, we assume that, there is a postgres database deployed on `Kubernetes` already. 
+You can also modify the database configuration to use a postgres database outside the `Kubernetes` cluster.
+Deploy the index `Flow` on `Kubernetes`:
+```python
+from jina import Flow
+
+flow = Flow(
+    name='index-flow', port_expose=8080, infrastructure='K8S', protocol='http'
+).add(
+    # name of the service and deployment in Kubernetes
+    name='test_searcher',
+    # executor has to be containerized
+    uses='jinahub+docker://FaissPostgresSearcher',
+    # database configuration
+    uses_with={
+        'hostname': f'postgres.postgres.svc.cluster.local',
+        'username': 'postgresadmin',
+        'database': 'postgresdb',
+        'table': 'test_searcher',
+    }
+)
+flow.start()
+```
+Use `port-forward` to send requests to the gateway of the index `Flow`.
```

Comment:
Thats not a Jina thing, but should we emphasize that this is a dev/debug mechanism and not meant for production? And that the user has to worry about how to expose this publicly in a proper way? We can add some links to the necessary K8s concepts. 
On the other hand this makes the doc unnecessary long.

Reply:
It is in the experimental section. Do you think that is enough? Or should I add a warning as well?